modules/programs/kitty.nix

```diff
@@ -111,48 +134,87 @@ in {
     };
   };
 
-  config = mkIf cfg.enable {
-    home.packages = [ cfg.package ] ++ optionalPackage cfg.font;
-
-    xdg.configFile."kitty/kitty.conf" = {
-      text = ''
-        # Generated by Home Manager.
-        # See https://sw.kovidgoyal.net/kitty/conf.html
-      '' + concatStringsSep "\n" ([
-
-        (optionalString (cfg.font != null) ''
-          font_family ${cfg.font.name}
-          ${optionalString (cfg.font.size != null)
-          "font_size ${toString cfg.font.size}"}
-        '')
-
-        (optionalString (cfg.theme != null) ''
-          include ${pkgs.kitty-themes}/${
-            (head (filter (x: x.name == cfg.theme) (builtins.fromJSON
-              (builtins.readFile "${pkgs.kitty-themes}/themes.json")))).file
-          }
-        '')
-        (toKittyConfig cfg.settings)
-        (toKittyKeybindings cfg.keybindings)
-        (toKittyEnv cfg.environment)
-        cfg.extraConfig
-      ]);
-    } // optionalAttrs pkgs.stdenv.hostPlatform.isLinux {
-      onChange = ''
-        ${pkgs.procps}/bin/pkill -USR1 -u $USER kitty || true
-      '';
-    };
-
-    xdg.configFile."kitty/macos-launch-services-cmdline" = mkIf
-      (cfg.darwinLaunchOptions != null && pkgs.stdenv.hostPlatform.isDarwin) {
-        text = concatStringsSep " " cfg.darwinLaunchOptions;
+  config = mkIf cfg.enable (mkMerge [
+    {
+      home.packages = [ cfg.package ] ++ optionalPackage cfg.font;
+
+      xdg.configFile."kitty/kitty.conf" = {
+        text = ''
+          # Generated by Home Manager.
+          # See https://sw.kovidgoyal.net/kitty/conf.html
+        '' + concatStringsSep "\n" ([
+
+          (optionalString (cfg.font != null) ''
+            font_family ${cfg.font.name}
+            ${optionalString (cfg.font.size != null)
+            "font_size ${toString cfg.font.size}"}
+          '')
+
+          (optionalString (cfg.theme != null) ''
+            include ${pkgs.kitty-themes}/${
+              (head (filter (x: x.name == cfg.theme) (builtins.fromJSON
+                (builtins.readFile "${pkgs.kitty-themes}/themes.json")))).file
+            }
+          '')
+
+          (optionalString (cfg.shellIntegration.mode != "disabled") ''
+            # Shell integration is sourced and configured manually
+            shell_integration disabled
+          '')
+          (toKittyConfig cfg.settings)
+          (toKittyKeybindings cfg.keybindings)
+          (toKittyEnv cfg.environment)
+          cfg.extraConfig
+        ]);
+      } // optionalAttrs pkgs.stdenv.hostPlatform.isLinux {
+        onChange = ''
+          ${pkgs.procps}/bin/pkill -USR1 -u $USER kitty || true
+        '';
       };
-  };
+
+      xdg.configFile."kitty/macos-launch-services-cmdline" = mkIf
+        (cfg.darwinLaunchOptions != null && pkgs.stdenv.hostPlatform.isDarwin) {
+          text = concatStringsSep " " cfg.darwinLaunchOptions;
+        };
+    }
+    (mkIf (cfg.shellIntegration.mode != "disabled") {
```

Comment:
This would silently ignore the `enable*Integration` options by default, until the user figures out they also need to set `shellIntegration.mode` to `enabled`. (Why isn't that the default?)

Reply:
The default values of `enable*Integration` are now `config.programs.kitty.shellIntegration.mode != "disabled"`, which should be a bit less confusing?